lib/reverse_sentence.rb

```diff
@@ -1,6 +1,50 @@
 # A method to reverse the words in a sentence, in place.
 # Time complexity: ?
+# n = mysentence length
+# m = newsentence length
+# n + 1/2m + m =  n + 3/2m => O(n + m) (1/2 n b/c since it's decrementing and increamenting at the same time)
+
 # Space complexity: ?
+# n = new_arr length == my_sentence
+# m = holder length stores only words without space,
+# p = new_sentence
+# n + m + p => O(n + m + p)
+
 def reverse_sentence(my_sentence)
```

Comment:
I would say this is O(n) space complexity, where n is the number of letters.  This is because the `new_arr` is an array which totals all the individual words.

The time complexity would be O(n) since the longest loop here has to run through all the letters in the string.

 There is an alternative solution which is also O(n) time complexity and O(1) space complexity.

Consider: Reverse the string completely. Then loop through the string reversing each individual word, in place (i.e. not making an array or another string). Think about how you might be able to do that.